### PR TITLE
Strict type `Dependabot::Hex::MixFileSanitizer`

### PR DESCRIPTION
**Pull request by raj-meka**
### What are you trying to accomplish?
Sorbet Implementation for Hex MixFileSanitizer ruby file.

### Anything you want to highlight for special attention from reviewers?
Please pay attention to the fixes in the Hex MixFileSanitizer ruby file.

### How will you know you've accomplished your goal?
The sorbet rules will fire as needed.

### Checklist

- [x]  I have run the complete test suite to ensure all tests and linters pass.
- [x]  I have thoroughly tested my code changes to ensure they work as expected, including adding additional tests for new functionality.
- [x]  I have written clear and descriptive commit messages.
- [x]  I have provided a detailed description of the changes in the pull request, including the problem it addresses, how it fixes the problem, and any relevant details about the implementation.
- [x]  I have ensured that the code is well-documented and easy to understand.

### CODE REVIEW

File: hex/lib/dependabot/hex/file_updater/mixfile_sanitizer.rb [hunk 1]
```diff
@@ -1,50 +1,56 @@
...
 
         private
 
-        attr_reader :mixfile_content
-
-        def prevent_version_file_loading(configuration)
+        sig { params(configuration: String).returns(String) }
+        def prevent_config_path_loading(configuration)
           configuration
+            .gsub(/^\s*config_path:.*(?:,|$)/, "")
```
Comment (abdulapopoola) [inline hex/lib/dependabot/hex/file_updater/mixfile_sanitizer.rb:40]: should this be removed?

File: hex/lib/dependabot/hex/file_updater/mixfile_sanitizer.rb [hunk 2]
```diff
@@ -1,50 +1,56 @@
...
             .gsub(NESTED_VERSION_FILE_READ_BANG, 'String.trim("0.0.1")')
```
Comment (abdulapopoola) [inline hex/lib/dependabot/hex/file_updater/mixfile_sanitizer.rb:41]: why is this included?

File: hex/lib/dependabot/hex/file_updater/mixfile_sanitizer.rb [hunk 3]
```diff
@@ -1,51 +1,53 @@
...
-            .then(&method(:prevent_config_path_loading))
+          @mixfile_content
+            .then { |content| prevent_version_file_loading(content) }
+            .then { |content| prevent_config_path_loading(content) }
         end
-        # rubocop:enable Performance/MethodObjectAsBlock
 
         private
 
-        attr_reader :mixfile_content
```
Comment (abdulapopoola) [inline hex/lib/dependabot/hex/file_updater/mixfile_sanitizer.rb:35]: can this be added back?

**Review by abdulapopoola — APPROVED** [review]

**Review by JamieMagee — APPROVED** [review]
